src/main/java/com/commercetools/sync/categories/helpers/CategorySyncOptions.java

```diff
@@ -0,0 +1,89 @@
+        super(ctpProjectKey, ctpClientId, ctpClientSecret);
+        this.updateActionsFilter = updateActionsFilter;
+    }
+
+    public CategorySyncOptions(@Nonnull final SphereClientConfig sphereClientConfig,
+                               @Nonnull final Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> updateActionsFilter) {
+        super(sphereClientConfig);
+        this.updateActionsFilter = updateActionsFilter;
+    }
+
+    public CategorySyncOptions(@Nonnull final String ctpProjectKey,
+                               @Nonnull final String ctpClientId,
+                               @Nonnull final String ctpClientSecret,
+                               @Nonnull final BiConsumer<String, Throwable> updateActionErrorCallBack,
+                               @Nonnull final Consumer<String> updateActionWarningCallBack) {
+        super(ctpProjectKey, ctpClientId, ctpClientSecret, updateActionErrorCallBack, updateActionWarningCallBack);
+    }
+
+    public CategorySyncOptions(@Nonnull final SphereClientConfig sphereClientConfig,
+                               @Nonnull final BiConsumer<String, Throwable> updateActionErrorCallBack,
+                               @Nonnull final Consumer<String> updateActionWarningCallBack) {
+        super(sphereClientConfig, updateActionErrorCallBack, updateActionWarningCallBack);
+    }
+
+    public CategorySyncOptions(@Nonnull final String ctpProjectKey,
+                               @Nonnull final String ctpClientId,
+                               @Nonnull final String ctpClientSecret,
+                               @Nonnull final BiConsumer<String, Throwable> updateActionErrorCallBack,
+                               @Nonnull final Consumer<String> updateActionWarningCallBack,
+                               @Nonnull final Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> updateActionsFilter) {
+        super(ctpProjectKey, ctpClientId, ctpClientSecret, updateActionErrorCallBack, updateActionWarningCallBack);
+        this.updateActionsFilter = updateActionsFilter;
+    }
+
+    public CategorySyncOptions(@Nonnull final SphereClientConfig sphereClientConfig,
+                               @Nonnull final BiConsumer<String, Throwable> updateActionErrorCallBack,
+                               @Nonnull final Consumer<String> updateActionWarningCallBack,
+                               @Nonnull final Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> updateActionsFilter) {
+        super(sphereClientConfig, updateActionErrorCallBack, updateActionWarningCallBack);
+        this.updateActionsFilter = updateActionsFilter;
+    }
+
+    public Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> getUpdateActionsFilter() {
+        return updateActionsFilter;
+    }
+
+    public CategoryService getCategoryService() {
+        if (categoryService == null) {
```

Comment:
but this constructor is exactly what I mean by 

> I intentionally didn't force injection, because I want the services not to be initialised by the user of the sync, but rather by the sync itself.

The users should only be able to initialise the sync from this constructor. The other constructor is for testing purposes. ( I add doc about it:  3cc840f)